### PR TITLE
Collections image optimization

### LINKED ISSUE #784: Improve loading performance of images on collections list page
#### Description
This is more of an improvement than a bug. 

Full sized images are being requested and loaded for the collection images on the [list page](https://www.napari-hub.org/collections). For example, [this image](https://www.napari-hub.org/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Fchanzuckerberg%2Fnapari-hub-collections%2Fmain%2Fimages%2Fnational-cancer-institute-evhLgfOjU5Y-unsplash.jpg&w=3840&q=75) is 3840x3840 pixels but we are cropping it to be visible for 225x100 pixels only so we shouldn't load the full image. 

The first time the page loaded, it took several seconds (955ms to 5.2s) to load all the images. Subsequent loads took less time because of the cache, but still more than necessary.

#### Expected Results
Collection images should load in under a second.

#### Actual Results
Collection images are currently taking several seconds to load on initial load.

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

Closes #784

This improves image loading for the collections index page by using a smaller sized thumbnail image instead of the full cover image. This reduces the average uncached load time from 1s - 15s to  300ms - 600ms

This PR depends on https://github.com/chanzuckerberg/napari-hub-collections/pull/37 to be merged because it includes the thumbnail images.

## Demos

https://dev-fast-images-frontend.dev.imaging.cziscience.com/collections

### Uncached Before and After

| Before                                                                                                                                   | After                                                                                                                                    |
| ---------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
| image | image |

### Cached Before and After

| Before                                                                                                                                   | After                                                                                                                                    |
| ---------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
| image | image |

### CODE REVIEW

File: backend/api/collections.py [hunk 1]
```diff
@@ -27,7 +27,12 @@ def get_yaml_data(collection_name, visibility_requirements):
     if yaml_file:
         data = yaml.safe_load(yaml_file)
         if data and data.get("visibility", "public") in visibility_requirements:
-            data["cover_image"] = IMAGES_BASE_URL + data.get("cover_image")
+            cover_image = data.get("cover_image")
+            ext = cover_image.split('.')[-1]
+            thumb_image = cover_image.replace(f'.{ext}', f'-thumb.{ext}')
+
+            data["cover_image"] = IMAGES_BASE_URL + cover_image
+            data["thumb_image"] = IMAGES_BASE_URL + thumb_image
```
Comment (manasaV3) [inline backend/api/collections.py:35]: More of a thought for the future implementation. I understand we need to do this in the backend at the moment, as we are using GitHub files as data sources. 
But, once we move that to a more persistent data store, I think we should add this to the data when we save it in our systems.

**Review by manasaV3 — DISMISSED** [review]
:shipit:

File: frontend/package.json [hunk 2]
```diff
@@ -106,6 +108,7 @@
     "remark-rehype": "10.1.0",
     "remark-remove-comments": "0.2.0",
     "remark-slug": "7.0.1",
+    "sharp": "^0.31.2",
```
Comment (richaagarwal) [inline frontend/package.json:111]: Curious where this gets used?

File: frontend/src/components/CollectionsPage/CollectionCard.tsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ export function CollectionCard({ collection }: Props) {
       <div className="relative w-full h-[50px] screen-495:h-[100px]">
         <Image
           className="object-cover"
-          src={collection.cover_image}
+          src={collection.thumb_image}
           alt={`${collection.title} cover image`}
```
Comment (richaagarwal) [inline frontend/src/components/CollectionsPage/CollectionCard.tsx:31]: super minor but should we update the alt text as well?

**Review by richaagarwal — APPROVED** [review]

**Pull request merged**